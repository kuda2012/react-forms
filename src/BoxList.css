.BoxList {
  display: flex;
  flex-direction: row;
  width: 60%;
  justify-content: space-between;
  border: solid black 2px;
  flex-wrap: wrap;
  align-self: center;
}
