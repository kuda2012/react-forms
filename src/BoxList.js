import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import "./BoxList.css";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
  //   const INITIAL_STATE = [
  //     { id: uuid(), width: "200px", height: "200px", backgroundColor: "blue" },
  //     { id: uuid(), width: "200px", height: "400px", backgroundColor: "green" },
  //   ];
  const [boxes, setBoxes] = useState([]);

  const addBox = (newBox) => {
    setBoxes((boxes) => [...boxes, { ...newBox, id: uuid() }]);
  };
  const deleteBox = (id) => {
    setBoxes((boxes) => {
      return boxes.filter((box) => box.id !== id);
    });
  };

  return (
    <>
      <h3>Box List</h3>
      <NewBoxForm addBox={addBox} />
      {boxes.length !== 0 && (
        <div className="BoxList" data-testid="BoxList">
          {boxes.map((box, i) => (
            <Box
              number={i}
              key={box.id}
              id={box.id}
              width={box.width}
              height={box.height}
              backgroundColor={box.backgroundColor}
              deleteBox={deleteBox}
            />
          ))}
        </div>
      )}
    </>
  );
};

export default BoxList;
