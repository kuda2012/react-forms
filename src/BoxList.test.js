import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function () {
  render(<BoxList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});
it("shows title", function () {
  const { queryByText } = render(<BoxList />);
  const title = queryByText("Box List");
  expect(title).toBeInTheDocument();
});
it("should make a new box", () => {
  const { queryByLabelText, queryByText, queryByTestId } = render(<BoxList />);
  const BoxListBefore = queryByTestId("Box1");
  expect(BoxListBefore).not.toBeInTheDocument();
  const boxBeforeCreation = queryByTestId("Box1");
  expect(boxBeforeCreation).not.toBeInTheDocument();
  const width = queryByLabelText("Width");
  const height = queryByLabelText("Height");
  const color = queryByLabelText("Color");
  const btn = queryByText("Create Box");
  fireEvent.change(width, { target: { value: "300px" } });
  fireEvent.change(height, { target: { value: "300px" } });
  fireEvent.change(color, { target: { value: "blue" } });
  fireEvent.click(btn);
  const BoxListAfter = queryByTestId("BoxList");
  expect(BoxListAfter).toBeInTheDocument();
  const boxCreated = queryByTestId("Box0");
  expect(boxCreated).toBeInTheDocument();
});

it("should delete a box", () => {
  const { queryByLabelText, queryByText, queryByTestId } = render(<BoxList />);
  const BoxListBefore = queryByTestId("Box1");
  expect(BoxListBefore).not.toBeInTheDocument();
  const boxBeforeCreation = queryByTestId("Box1");
  expect(boxBeforeCreation).not.toBeInTheDocument();
  const width = queryByLabelText("Width");
  const height = queryByLabelText("Height");
  const color = queryByLabelText("Color");
  const btn = queryByText("Create Box");
  fireEvent.change(width, { target: { value: "300px" } });
  fireEvent.change(height, { target: { value: "300px" } });
  fireEvent.change(color, { target: { value: "blue" } });
  fireEvent.click(btn);
  const BoxListAfter = queryByTestId("BoxList");
  expect(BoxListAfter).toBeInTheDocument();
  const boxCreated = queryByTestId("Box0");
  expect(boxCreated).toBeInTheDocument();
  const deleteBox = queryByText("X");
  fireEvent.click(deleteBox);
  expect(BoxListAfter).not.toBeInTheDocument();
  expect(boxCreated).not.toBeInTheDocument();
});
