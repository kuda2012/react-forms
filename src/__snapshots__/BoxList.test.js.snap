// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h3>
    Box List
  </h3>
  <form>
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      name="width"
      placeholder="Box Width ex:200px "
      type="text"
      value="200px"
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      name="height"
      placeholder="Box Height ex:200px "
      type="text"
      value="200px"
    />
    <label
      for="backgroundColor"
    >
      Color
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      placeholder="Box Color"
      type="text"
      value=""
    />
    <button>
      Create Box
    </button>
  </form>
</DocumentFragment>
`;
